### PR TITLE
Add Intel specific definitions from https://github.com/KhronosGroup/S…

### PR DESCRIPTION
**Pull request by MrSidims**
…PIRV-LLVM-Translator

List of extensions:
SPV_INTEL_fpga_memory_attributes
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_fpga_memory_attributes.asciidoc

SPV_INTEL_kernel_attributes
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_kernel_attributes.asciidoc

SPV_INTEL_fpga_reg
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_fpga_reg.asciidoc

SPV_INTEL_blocking_pipes
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_blocking_pipes.asciidoc

SPV_INTEL_fpga_loop_controls
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_fpga_loop_controls.asciidoc

SPV_INTEL_unstructured_loop_controls
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_unstructured_loop_controls.asciidoc

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
There are several definitions, that are not presenting in spirv.core.grammar.json, like RayFlagsShift which were removed after headers generation. Was it to be expected?

File: include/spirv/unified1/spirv.core.grammar.json [hunk 1]
```diff
@@ -6982,6 +6982,59 @@
...
+        { "kind" : "LiteralInteger", "quantifier" : "*", "name" : "'Loop Control Parameters'" }
+      ],
+      "capabilities" : [ "UnstructuredLoopControlsINTEL" ],
+      "extensions" : [ "SPV_INTEL_unstructured_loop_controls" ],
+      "version" : "None"
+    },
+    {
+      "opname" : "OpReadPipeBlockingINTEL",
+      "class"  : "Pipe",
+      "opcode" : 5946,
```
Comment (johnkslang) [inline include/spirv/unified1/spirv.core.grammar.json:7000]: I'm concerned about continuing to create gaps here.  See issue #151.  I think you have all the intervening consecutive opcodes reserved.

Comment (MrSidims) [reply include/spirv/unified1/spirv.core.grammar.json:7000]: The gaps will be partially filled when https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/SPIRV/SPV_INTEL_function_pointers.asciidoc and https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/SPIRV/SPV_INTEL_inline_assembly.asciidoc are published here.

Comment (MrSidims) [reply include/spirv/unified1/spirv.core.grammar.json:7000]: Nevermind, this particular gap won't be filled by these extensions.

**Comment by MrSidims** [comment]
Left only json changes excluding generated headers.

**Comment by johnkslang** [comment]
Could you revisit this?  Look at avoiding gaps (there was a comment about that), resolve conflicts.  Did you want me to generated the headers?

**Comment by MrSidims** [comment]
> Could you revisit this? Look at avoiding gaps (there was a comment about that), resolve conflicts. Did you want me to generated the headers?

I'll take a look what can be done to avoid the gaps now. The problem here is that the values present in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/lib/SPIRV/libSPIRV/spirv.hpp as is.

Regarding the headers generation - I can do it myself, was just a little bit confused, that several definitions disappeared, see my comment from above.

**Comment by johnkslang** [comment]
I just rebuilt headers from your other commit, a few minutes ago.  It all seemed consistent and okay.

It is okay if I build headers.

One thing we want to do is reuse enumerant values (from other enums) for the opcodes.  If you have any reserved that have been used for other enums, feel free to reuse them for instructions also.

**Comment by MrSidims** [comment]
Got it. Currently we are filling the op codes interval from 5568 to 5631 with following extensions: 
https://github.com/intel/llvm/pull/1593
https://github.com/intel/llvm/pull/1611
https://github.com/intel/llvm/pull/1612

will come to the mentioned in this PR gap in the future.

File: include/spirv/unified1/spirv.core.grammar.json [hunk 2]
```diff
@@ -6688,6 +6688,71 @@
       "capabilities" : [ "SubgroupAvcMotionEstimationINTEL" ],
       "version" : "None"
     },
+    {
+      "opname" : "OpSubgroupAvcSicGetInterRawSadsINTEL",
+      "class"  : "@exclude",
+      "opcode" : 5816,
```
Comment (johnkslang) [inline include/spirv/unified1/spirv.core.grammar.json:6694]: It seems this created two `OpSubgroupAvcSicGetInterRawSadsINTEL`.  Is one supposed to be an alias, or we can have just one?

File: include/spirv/unified1/spirv.core.grammar.json [hunk 3]
```diff
@@ -11226,6 +11509,48 @@
...
+        },
+        {
+          "enumerant" : "FPGARegINTEL",
+          "value" : 5948,
+          "extensions" : [ "SPV_INTEL_fpga_reg" ],
+          "version" : "None"
+        },
+        {
+          "enumerant" : "KernelAttributesINTEL",
+          "value" : 5892,
```
Comment (johnkslang) [inline include/spirv/unified1/spirv.core.grammar.json:11545]: We want to keep these in numerical order.

File: include/spirv/unified1/spirv.core.grammar.json (hunk 2 above)
Comment (MrSidims) [reply include/spirv/unified1/spirv.core.grammar.json:6694]: I suppose it's an artifact appeared after rebase.

File: include/spirv/unified1/spirv.core.grammar.json (hunk 3 above)
Comment (MrSidims) [reply include/spirv/unified1/spirv.core.grammar.json:11545]: Oops, thanks!

**Comment by MrSidims** [comment]
BTW, what are the rules for updating PRs in this repository? Shall fixes be amended into the original commit before updating, or it's OK to have commits, like: "Apply suggestions" and this commits will be squashed during the merge?

**Comment by johnkslang** [comment]
Something like this should go into the repo as a single commit.  I'll squash when I remember.  It is sometimes important to see feedback addressed as a separate commit, just for efficiency, but better to not have that jitter in master history.

Force pushing the PR is okay, when you want to control how the actual history will look.

This one is okay as it is now.

**Pull request merged**